### PR TITLE
Source Greenhouse: unpin CDK

### PR DESCRIPTION
**Pull request by ChristoGrab**
## What

Unpinning CDK version for source Greenhouse.

## How

- Updated poetry dependencies and manifest version
- Small change to the StreamSlicer custom component to return a StreamSlice (instead of a dict) to account for changes made to the CDK in [0.68.2](https://github.com/airbytehq/airbyte/pull/35471)

### CODE REVIEW

File: airbyte-integrations/connectors/source-greenhouse/pyproject.toml
Comment (bazarnov) [inline airbyte-integrations/connectors/source-greenhouse/pyproject.toml:20]: it's better to use ">=0.63.2", but not blocking on this one.

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py
Comment (bazarnov) [inline airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py:112]: What if there is no `parent_state_value`?

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py
Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py]: Took another look, and now realizing the variable name is a bit misleading. This is set to the value of the current parent record's primary key, so we can assume it always exists. I've updated the variable name to `parent_primary_key` for clarity.

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py
Comment (brianjlai) [inline airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py:28]: nit: if we're only yielding one slice in the subsequent line, can we just do this in one line instead

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py
Comment (brianjlai) [inline airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py:115]: remove extra print statement

**Review by brianjlai — APPROVED** [review]
some nits. In particular please get rid of the extra print statement before merging. But don't need to block on approval

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py
Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py]: Done 👍

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py
Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py]: Oh whoops! Thanks for catching, removed

**Pull request merged**
